pkg/ccl/changefeedccl/rowfetcher_cache.go

```diff
@@ -173,6 +173,10 @@ func (c *rowFetcherCache) RowFetcherForTableDesc(
 	); err != nil {
 		return nil, err
 	}
+
+	// Necessary because virtual columns are not populated.
+	rf.IgnoreUnexpectedNulls = true
+
```

Comment:
Is it possible to do it only if we know that we are hanging virtual non null columns?